src/KubeLibrary/KubeLibrary.py

```diff
@@ -119,11 +165,18 @@ def reload_config(self, kube_config=None, context=None, api_url=None, bearer_tok
             configuration.host = api_url
             configuration.ssl_ca_cert = ca_cert
             self.api_client = client.ApiClient(configuration)
+            configuration.verify_ssl = cert_validation
```

Comment:
I think you should move 
```
                configuration.verify_ssl = cert_validation
                self.dynamic_client = dynamic.DynamicClient(client.ApiClient(configuration=configuration))
```

to self._add_api(...), so that there would sth like
```
self._add_api('dynamic', dynamic.DynamicClient)
```
then move all the disabling of cert validation in _add_api, this way it will follow existing initialization and be more DRY.

Reply:
Yes very good point. I will post my initial attempt below. If you generally agree with that, I can commit this.

```
     class DynamicClient(dynamic.DynamicClient):
    @property
    def api_client(self):
        return self.client
     
   ....

  def reload_config(self, kube_config=None, context=None, api_url=None, bearer_token=None, ca_cert=None, incluster=False, cert_validation=True):
        
        self.api_client = None
        self.cert_validation = cert_validation
        self.dynamic = None
        if incluster:
            try:
                config.load_incluster_config()
                self.api_client = client.ApiClient(configuration=client.Configuration().get_default_copy())
            except config.config_exception.ConfigException as e:
                logger.error('Are you sure tests are executed from within k8s cluster?')
                raise e
        elif api_url and bearer_token:
            if bearer_token.startswith('Bearer '):
                raise BearerTokenWithPrefixException
            configuration = client.Configuration()
            configuration.api_key["authorization"] = bearer_token
            configuration.api_key_prefix['authorization'] = 'Bearer'
            configuration.host = api_url
            configuration.ssl_ca_cert = ca_cert
            self.api_client = client.ApiClient(configuration)
        else:
            try:
                config.load_kube_config(kube_config, context)
                self.api_client = config.new_client_from_config(config_file=kube_config, context=context)
            except TypeError:
                logger.error('Neither KUBECONFIG nor ~/.kube/config available.')

        self._add_api('v1', client.CoreV1Api)
        self._add_api('extensionsv1beta1', client.ExtensionsV1beta1Api)
        self._add_api('batchv1', client.BatchV1Api)
        self._add_api('appsv1', client.AppsV1Api)
        self._add_api('batchv1_beta1', client.BatchV1beta1Api)
        self._add_api('custom_object', client.CustomObjectsApi)
        self._add_api('rbac_authv1_api', client.RbacAuthorizationV1Api)
        self._add_api('autoscalingv1', client.AutoscalingV1Api)
        self._add_api('dynamic', DynamicClient)

     ....

```
PS: I ended up leaving the `self.api_client` initializations under `reload_config` since the dynamic client is not able to deal with `None` clients. If you got a better idea, I can try to implement that.

Reply:
Didn't have time to test it myself, but from what I checked both load_incluster_config and load_kube_config sets default config, which means you could just add:
```
if( not self.api_client)
    self.api_client = client.ApiClient(configuration=client.Configuration().get_default_copy())

 self._add_api('v1', client.CoreV1Api)
        self._add_api('extensionsv1beta1', client.ExtensionsV1beta1Api)
(...)
```

This way we would always have api_client in place before _add_api block.